src/StreamableFeesLocker.sol

```diff
@@ -0,0 +1,340 @@
+        params[1] = abi.encode(poolKey.currency0, poolKey.currency1, address(this));
+
+        positionManager.modifyLiquidities(abi.encode(actions, params), block.timestamp);
+
+        BeneficiaryData[] memory beneficiaries = position.beneficiaries;
+
+        (uint256 currency0ToDistribute, uint256 currency1ToDistribute) = _updateCurrencyBalances(position, poolKey);
+
+        uint256 amount0Distributed;
+        uint256 amount1Distributed;
+        address beneficiary;
+        for (uint256 i; i < beneficiaries.length; ++i) {
+            beneficiary = beneficiaries[i].beneficiary;
+            uint256 shares = beneficiaries[i].shares;
+
+            // Calculate share of fees for this beneficiary
+            uint256 amount0 = currency0ToDistribute * shares / WAD;
+            uint256 amount1 = currency1ToDistribute * shares / WAD;
+
+            _distributeFees(beneficiary, poolKey, amount0, amount1);
+
+            amount0Distributed += amount0;
+            amount1Distributed += amount1;
+        }
+
+        // Distribute the remaining fees to the last beneficiary
+        uint256 amount0Remaining =
+            currency0ToDistribute > amount0Distributed ? currency0ToDistribute - amount0Distributed : 0;
+        uint256 amount1Remaining =
+            currency1ToDistribute > amount1Distributed ? currency1ToDistribute - amount1Distributed : 0;
+        _distributeFees(beneficiary, poolKey, amount0Remaining, amount1Remaining);
+
+        // Note: For no-op governance, if recipient is DEAD_ADDRESS (0xdead), the position will be permanently locked
+        // and beneficiaries can collect fees in perpetuity
+        if (block.timestamp >= position.startDate + position.lockDuration && position.recipient != DEAD_ADDRESS) {
+            position.isUnlocked = true;
+
+            // Transfer the position to the recipient
+            ERC721(address(positionManager)).safeTransferFrom(address(this), position.recipient, tokenId, new bytes(0));
+
+            emit Unlock(tokenId, position.recipient);
+        }
+
+        // Update the position in storage
+        positions[tokenId] = position;
+
+        emit DistributeFees(tokenId, currency0ToDistribute, currency1ToDistribute);
+    }
+
+    /// @notice Releases accrued fees to the caller
+    /// @param tokenId ID of the position to release fees from
+    function releaseFees(
+        uint256 tokenId
+    ) external nonReentrant {
+        // Check if position exists
+        PositionData memory position = positions[tokenId];
+        require(position.startDate != 0, PositionNotFound());
+
+        // Check if sender is a beneficiary
+        bool isBeneficiary = false;
+        for (uint256 i = 0; i < position.beneficiaries.length; i++) {
+            if (position.beneficiaries[i].beneficiary == msg.sender) {
+                isBeneficiary = true;
+                break;
+            }
+        }
+        require(isBeneficiary, NotBeneficiary());
```

Comment:
Gas: I think this check is redundant since we only release fees accounted to the `msg.sender` in `_releaseFees`

Reply:
i think we will probably keep this as is so we can terminate early with a proper error